middle_end/simple_value_approx.ml

```diff
@@ -392,6 +416,8 @@ let useful t =
   | Value_float _ | Value_boxed_int _ | Value_closure _ | Value_extern _
   | Value_symbol _ -> true
 
+let all_not_useful ts = List.for_all (fun t -> not (useful t)) ts
```

Comment:
We may need to think a bit about that. I can't see a lot of meaningful examples right now, but there are some cases where some optimisation can happen even if we don't know anything about the shape of the arguments.

For instance

```
let f x y = x

let g x =
  let y = (x,x) in
  f x y
```

```
let f x y =
  if x = y then ... else ...

let g x = f x x
```